api/src/main/java/org/openmrs/module/fhir2/api/FhirService.java

```diff
@@ -11,12 +11,17 @@
 
 import javax.validation.constraints.NotNull;
 
+import java.util.List;
+import java.util.Set;
+
 import org.hl7.fhir.instance.model.api.IAnyResource;
 
 public interface FhirService<T extends IAnyResource> {
 	
 	T get(@NotNull String uuid);
 	
+	List<T> get(@NotNull Set<String> distinctUUIDs);
```

Comment:
Does this *need* to take a set? Could we instead support a list, array, or even generic collection?

Reply:
It can have other types as well, but I chose a set so that we only supply distinct uuids. To make the list as unique, we will anyway need to put it in a set, why not just pass that set directly. What do you think?

Reply:
So, while we want only the unique list of results, we can also defer the uniqueness part to the database (`select * from Concept where uuid in ('e110793d-939b-4661-82c1-b613883ea08d')` produces exactly the same output as `select * from Concept where uuid in ('e110793d-939b-4661-82c1-b613883ea08d', 'e110793d-939b-4661-82c1-b613883ea08d')`).

But my real thinking here was that if we build a method that takes a `Collection` instead of a `Set`, we can still continue to pass in a set, but we also gain some flexibility in usage (so that if someone has constructed a List, etc., they don't wind up having to convert it to a Set type).

Reply:
Yes, that makes sense. I'll change it to collection.